### PR TITLE
Switch to tag based release

### PR DESCRIPTION
**Pull request by dschwoerer**

### CODE REVIEW

File: .github/workflows/python_publish.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        with:
+          python-version: "3.x"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools wheel twine
+      - name: Build and publish
+        env:
+          TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```
Comment (ZedThree) [inline .github/workflows/python_publish.yml:24]: An alternative is to use the PYPA's `pypi-publish` action, and use a PyPI token scoped to just this package. See here for an example: https://github.com/pyro-kinetics/pyrokinetics/blob/unstable/.github/workflows/python_publish.yml

**Review by ZedThree — COMMENTED** [review]
Probably also want to add `_version.py` to `.gitignore`

**Comment by ZedThree** [comment]
Do we also want to move to using `setup.cfg` instead of `setup.py`?

File: .github/workflows/python_publish.yml (hunk 1 above)
Comment (dschwoerer) [reply .github/workflows/python_publish.yml:24]: I haven't looked at that action, but that should be able with this setup as well.

Just set `PYPI_USERNAME` to `__token__` and then the `PYPI_PASSWORD` to the token itself.

Comment (ZedThree) [reply .github/workflows/python_publish.yml:24]: I don't think it makes much of a difference, apart from being able to scope the token

**Comment by dschwoerer** [comment]
scm version says both `setup.cfg` and `setup.py` are deprecated and `pyproject.toml` should be used. Are there good reasons to switch to `setup.cfg` / `pyproject.toml`? I would use it for new projects, but am not sure whether it is worth to switch existing projects.

**Review by ZedThree — APPROVED** [review]

File: .github/workflows/python_publish.yml (hunk 1 above)
Comment (dschwoerer) [reply .github/workflows/python_publish.yml:24]: The token can anyway be scoped to just this project? That is at least what I do at https://github.com/dschwoerer/xemc3 - where I use the same action ...

I think only the first upload is problematic, as before that you cannot create a project specific token ...

**Pull request merged**